### ISSUE TITLE
Allow multiple orgs to access one API

### ISSUE THREAD

**Issue by altaurog** [TykTechnologies/tyk#2266]
**Do you want to request a *feature* or report a *bug*?**
I suppose a feature (perhaps a bug in documentation, though).

**What is the current behavior?**
Request is rejected if `org_id` in access policy does not match `org_id` in api definition, at least with OIDC auth.  This seems to be enforced on [two](https://github.com/TykTechnologies/tyk/blob/6a6e394b20503ded1cc57ac1a6d3e875fcb26190/mw_jwt.go#L605) distinct [levels](https://github.com/TykTechnologies/tyk/blob/9ddfe394f51d7b488a1c36cd21fbd3122baa313f/middleware.go#L240).  This is confusing, because the `providers.client_ids` and policy data structures seem to imply that different provider/client_ids could map to different policies with different `org_id`s, which all have access rights to the same api:

```
api 1-*> provider 1-*> client_id 1-*> policy (org_id) 1-*> api
```

I am not even sure I understand why `org_id` has to be specified on the api at all, given the above data structure, but even though the documentation [seems pretty clear that is is not required (only recommended?)](https://tyk.io/docs/tyk-rest-api/api-definition-objects/other-root-objects/), if it is not set, the equality test fails and the request is rejected with log message `attempting to apply policy from different organisation to key, skipping`.

Judging from the comment in the code explaining this behavior, “otherwise you could overwrite a session key with a policy from a different org!”  I imagine session keys are not segregated by org, which is puzzling.  Was that just an oversight?  Would setting `segregate_by_client` change that?

**What is the expected behavior?**
I should be able to leave `org_id` out of api definition entirely and define policies for multiple `org_id`s granting access to the api.

**Comment by altaurog** [comment]
Thinking about this some more, maybe I haven’t understood what the function of `org_id` is, and I would really like to see more documentation around this (I am getting 403 [where it seems it should be](https://tyk.io/docs/tyk-rest-api/organisation-quotas/)).  I discovered that I can assign the same `org_id` to multiple policies, so I could probably accomplish what I need for my current use case if I use the same `org_id` everywhere and assign different `access_rights` based only on the `client_id` -> `policy_id` mapping.
Also a puzzle I’d love to see more documentation around is the circular data structure I described above.  Why is the provider -> client_id -> policy map defined on the individual api level when the policies’ access_rights already point at the api?  Wouldn’t it have been enough to make the openid config global?
